app/javascript/components/student/OpenEditorButton.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { CopyToClipboardButton } from '../common'
+import { ComboButton } from '../common/ComboButton'
+import { StartExerciseButton } from './open-editor-button/StartExerciseButton'
+
+export const OpenEditorButton = (
+  props:
+    | { status: 'locked'; command: string; links: null }
+    | { status: 'available'; command: string; links: { start: string } }
+    | {
+        status: 'completed' | 'published'
+        command: string
+        links: { exercise: string }
+      }
+    | {
+        status: 'iterated' | 'started'
+        command: string
+        links: { exercise: string }
+      }
+): JSX.Element | null => {
+  switch (props.status) {
+    case 'locked':
+      return (
+        <ComboButton className="--disabled">
+          <ComboButton.PrimarySegment>
+            <div>Open editor</div>
+          </ComboButton.PrimarySegment>
+          <ComboButton.DropdownSegment>
+            <CopyToClipboardButton textToCopy={props.command} />
+          </ComboButton.DropdownSegment>
+        </ComboButton>
```

Comment:
This looks really nice!

Reply:
It looks like the dropdownsegment bit is the same for all the cases. Do you expect any differences there?

Reply:
I've extracted the dropdown segment bit. Thanks!